modules/advangelistsBidAdapter.js

```diff
@@ -0,0 +1,443 @@
+}
+
+function findAndFillParam(o, key, value) {
+  try {
+    if (typeof value === 'function') {
+      o[key] = value();
+    } else {
+      o[key] = value;
+    }
+  } catch (ex) {}
+}
+
+/*
+function createRenderer(bidRequest) {
+  const renderer = Renderer.install({
+    id: bidRequest.bidId,
+    url: OUTSTREAM_SRC,
+    loaded: false
+  });
+
+  renderer.setRender(outstreamRender);
+
+  return renderer;
+}
+
+function outstreamRender(bid) {
+  bid.renderer.push(() => {
+    window.Beachfront.Player(bid.adUnitCode, {
+      ad_tag_url: bid.vastUrl,
+      width: bid.width,
+      height: bid.height,
+      expand_in_view: false,
+      collapse_on_complete: true
+    });
+  });
+}
+*/
+function getOsVersion() {
+  let clientStrings = [
+    { s: 'Android', r: /Android/ },
+    { s: 'iOS', r: /(iPhone|iPad|iPod)/ },
+    { s: 'Mac OS X', r: /Mac OS X/ },
+    { s: 'Mac OS', r: /(MacPPC|MacIntel|Mac_PowerPC|Macintosh)/ },
+    { s: 'Linux', r: /(Linux|X11)/ },
+    { s: 'Windows 10', r: /(Windows 10.0|Windows NT 10.0)/ },
+    { s: 'Windows 8.1', r: /(Windows 8.1|Windows NT 6.3)/ },
+    { s: 'Windows 8', r: /(Windows 8|Windows NT 6.2)/ },
+    { s: 'Windows 7', r: /(Windows 7|Windows NT 6.1)/ },
+    { s: 'Windows Vista', r: /Windows NT 6.0/ },
+    { s: 'Windows Server 2003', r: /Windows NT 5.2/ },
+    { s: 'Windows XP', r: /(Windows NT 5.1|Windows XP)/ },
+    { s: 'UNIX', r: /UNIX/ },
+    { s: 'Search Bot', r: /(nuhk|Googlebot|Yammybot|Openbot|Slurp|MSNBot|Ask Jeeves\/Teoma|ia_archiver)/ }
+  ];
+  let cs = find(clientStrings, cs => cs.r.test(navigator.userAgent));
+  return cs ? cs.s : 'unknown';
+}
+
+function getFirstSize(sizes) {
+  return (sizes && sizes.length) ? sizes[0] : { w: undefined, h: undefined };
+}
+
+function parseSizes(sizes) {
+  return utils.parseSizesInput(sizes).map(size => {
+    let [ width, height ] = size.split('x');
+    return {
+      w: parseInt(width, 10) || undefined,
+      h: parseInt(height, 10) || undefined
+    };
+  });
+}
+
+function getVideoSizes(bid) {
+  return parseSizes(utils.deepAccess(bid, 'mediaTypes.video.playerSize') || bid.sizes);
+}
+
+function getBannerSizes(bid) {
+  return parseSizes(utils.deepAccess(bid, 'mediaTypes.banner.sizes') || bid.sizes);
+}
+
+function getTopWindowReferrer() {
+  try {
+    return window.top.document.referrer;
+  } catch (e) {
+    return '';
+  }
+}
+
+function getVideoTargetingParams(bid) {
+  return Object.keys(Object(bid.params.video))
+    .filter(param => includes(VIDEO_TARGETING, param))
+    .reduce((obj, param) => {
+      obj[ param ] = bid.params.video[ param ];
+      return obj;
+    }, {});
+}
+
+function createVideoRequestData(bid, bidderRequest) {
+  let topLocation = getTopWindowLocation(bidderRequest);
+  let topReferrer = getTopWindowReferrer();
+
+  let sizes = getVideoSizes(bid);
+  let firstSize = getFirstSize(sizes);
+
+  let video = getVideoTargetingParams(bid);
+  var o = {
```

Comment:
avoid `var`, use `const` since it is not reassigned

Reply:
Thanks. replaced var with const